### PR TITLE
Enable ESLint

### PR DESCRIPTION
**Pull request by captbaritone**
This pull request replaces the old "jsl" linter with ESLint and enables it as part of our integration tests.

So far, the only rules it includes are the ones which are both currently enforced in Underscore AND are already passing in Backbone.

I will follow up with individual pull requests for each rule which will require changes to the code in an attempt to bring us in sync with Underscore's style.

Reference: https://github.com/jashkenas/underscore/pull/2390#issuecomment-164858105 (@akre54)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,10 +18,10 @@
     "uglify-js": "^2.4.17"
   },
   "scripts": {
-    "test": "karma start && coffee test/model.coffee",
+    "test": "karma start && coffee test/model.coffee && npm run lint",
     "build": "uglifyjs backbone.js --mangle --source-map backbone-min.map -o backbone-min.js",
     "doc": "docco backbone.js && docco examples/todos/todos.js examples/backbone.localStorage.js",
-    "lint": "jsl -nofilelisting -nologo -conf docs/jsl.conf -process backbone.js"
+    "lint": "eslint underscore.js test/*.js"
```
Comment (craigmichaelmartin) [inline package.json]: backbone.js? :)

Comment (captbaritone) [inline package.json]: derp

**Comment by jridgewell** [comment]
:+1:, should we merge and wait for the extra PRs?

File: package.json [hunk 2]
```diff
@@ -17,10 +18,10 @@
     "uglify-js": "^2.4.17"
   },
   "scripts": {
-    "test": "karma start && coffee test/model.coffee",
+    "test": "karma start && coffee test/model.coffee && npm run lint",
```
Comment (akre54) [inline package.json]: Worth adding https://www.npmjs.com/package/karma-eslint ?

**Comment by akre54** [comment]
Awesome. Thanks so much for this

**Pull request merged**